specs/proto/consensus.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+// Define wire types for consensus messages.
+
+// https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#feeheader
+message FeeHeader {
+  uint64 base_rate = 1;
+  uint64 tip_rate = 2;
+}
+
+// https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#availabledataheader
+message AvailableDataHeader {
+  // array of 32-byte hashes
+  repeated bytes row_roots = 1;
+  // array of 32-byte hashes
+  repeated bytes col_roots = 2;
+}
+
+message ConsensusProposal {
+  uint64 height = 1;
+  uint64 round = 2;
+  uint64 timestamp = 3;
+  // 32-byte hash
+  bytes last_header_hash = 4;
+  // 32-byte hash
+  bytes last_commit_hash = 5;
+  // 32-byte hash
+  bytes consensus_root = 6;
+  FeeHeader fee_header = 7;
+  // 32-byte hash
+  bytes state_commitment = 8;
+  uint64 available_data_original_shares_used = 9;
+  AvailableDataHeader available_data_header = 10;
```

Comment:
Didn't you say `available_data_original_shares_used` wasn't necessary as it can be inferred from the `available_data_header`?

Reply:
Nono, the _square size_ can be inferred from the available data header. How many shares of that square are used still needs to be provided.

Reply:
Ah that makes sense!